syntax = "proto3";

import "google/protobuf/empty.proto";

package chat;

service ChatService {
    rpc SendMessage(Message) returns (google.protobuf.Empty) {}
    rpc CreateGroup(CreateGroupRequest) returns (Metadata) {}
    rpc GetListUser(ListUsersRequest) returns (ListUsersResponse) {}
    rpc FetchMetadata(google.protobuf.Empty) returns (Metadata) {}
    rpc AddUserToGroup(AddUserRequest) returns (Metadata) {}
}

message Message{
    int32 user_id = 1;
    string user_name = 2;
    string message = 3;
    string create_at = 4;
    int32 group_id = 5;
    int32 recipient_user_id = 6;
}

message CreateGroupRequest {
    int32 user_id = 1;
    string group_name = 2;
}

message Metadata {
    string metadata = 1;
}

message ListUsersRequest {
    int32 user_id = 1;
}

message ListUsersResponse {
    string user_list = 1;
}

message AddUserRequest {
    int32 user_id = 1;
    int32 group_id = 2;
}

